Src/LinearSolvers/MLMG/AMReX_MLNodeLap_3D_K.H

```diff
@@ -1135,6 +1135,263 @@ void mlndlap_gauss_seidel_aa (Box const& bx, Array4<Real> const& sol,
     });
 }
 
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void tridiagonal_solve (Array1D<Real,0,31>& a_ls, Array1D<Real,0,31>& b_ls, Array1D<Real,0,31>& c_ls,
+                        Array1D<Real,0,31>& r_ls, Array1D<Real,0,31>& u_ls, Array1D<Real,0,31>& gam,
+                        int ilen ) noexcept
+{
+    Real bet = b_ls(0);
+    u_ls(0) = r_ls(0) / bet;
+
+    for (int i = 1; i <= ilen - 1; i++) {
+        gam(i) = c_ls(i-1) / bet;
+        bet = b_ls(i) - a_ls(i)*gam(i);
+        if (bet == 0) amrex::Abort(">>>TRIDIAG FAILED");
+        u_ls(i) = (r_ls(i)-a_ls(i)*u_ls(i-1)) / bet;
+    }
+    for (int i = ilen-2; i >= 0; i--) {
+        u_ls(i) = u_ls(i) - gam(i+1)*u_ls(i+1);
+    }
+}
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void mlndlap_gauss_seidel_with_line_solve_aa (Box const& bx, Array4<Real> const& sol,
+                              Array4<Real const> const& rhs, Array4<Real const> const& sig,
+                              Array4<int const> const& msk,
+                              GpuArray<Real,AMREX_SPACEDIM> const& dxinv) noexcept
+{
+    Real facx = (1.0/36.0)*dxinv[0]*dxinv[0];
+    Real facy = (1.0/36.0)*dxinv[1]*dxinv[1];
+    Real facz = (1.0/36.0)*dxinv[2]*dxinv[2];
+    Real fxyz = facx + facy + facz;
+    Real fmx2y2z = -facx + 2.0*facy + 2.0*facz;
+    Real f2xmy2z = 2.0*facx - facy + 2.0*facz;
+    Real f2x2ymz = 2.0*facx + 2.0*facy - facz;
+    Real f4xm2ym2z = 4.0*facx - 2.0*facy - 2.0*facz;
+    Real fm2x4ym2z = -2.0*facx + 4.0*facy - 2.0*facz;
+    Real fm2xm2y4z = -2.0*facx - 2.0*facy + 4.0*facz;
+ 
+    const auto lo = amrex::lbound(bx);
+    const auto hi = amrex::ubound(bx);
+
+    int idir = -1;
+    int ilen = 33;
+
+    if ( (dxinv[0] <= dxinv[2]) and (dxinv[1] <= dxinv[2]) ) {
+        idir = 2;
+        ilen = hi.z - lo.z + 1;
+    }
+    if ( (dxinv[0] <= dxinv[1]) and (dxinv[2] <= dxinv[1]) ) {
+        idir = 1;
+        ilen = hi.y - lo.y + 1;
+    }
+    if ( (dxinv[1] <= dxinv[0]) and (dxinv[2] <= dxinv[0]) ) {
+        idir = 0;
+        ilen = hi.x - lo.x + 1;
+    }
+
+    if (ilen > 32) amrex::Abort("mlndlap_gauss_seidel_with_line_solve_aa is hard-wired to be no longer than 32");
+
+    Array1D<Real,0,31> a_ls,b_ls,c_ls,u_ls,r_ls,gam;
+
+
+    if ( idir == 2 ) 
+    {
+        for (int j = lo.y; j <= hi.y; ++j) 
+	{
+            AMREX_PRAGMA_SIMD
+            for (int i = lo.x; i <= hi.x; ++i) 
+	    {
```

Comment:
If you look this on github, the indention here and many other places are not right.  Please fix them (by avoid tabs).  It's better you fix it than someone else fixes it later, because it makes the history hard to follow.